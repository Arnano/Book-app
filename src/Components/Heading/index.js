import React from 'react';
import PropTypes from 'prop-types';
import { keys } from 'lodash';

import * as colors from '../../Style/colors';
import buildHeading from './style';

const Heading = ({ center, children, classes, color, id, level, size, weight }) => {
  const StyledHeading = buildHeading(level);
  return (
    <StyledHeading 
      center={center} 
      className={classes} 
      color={color} 
      id={id} 
      size={size} 
      weight={weight}>
        {children}
    </StyledHeading>
  );
};

Heading.defaultProps = {
  color: 'darkGrey',
  center: false
};

Heading.propTypes = {
  center: PropTypes.bool,
  children: PropTypes.node.isRequired,
  classes: PropTypes.string,
  color: PropTypes.oneOf([...keys(colors)]),
  id: PropTypes.string,
  level: PropTypes.oneOf([1, 2, 3, 4]).isRequired,
  size: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4']).isRequired,
  weight: PropTypes.oneOf(['fwSmall', 'fwMedium', 'fwLarge'])
};

export default Heading;
