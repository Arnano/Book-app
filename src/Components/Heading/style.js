import styled from 'styled-components';
import { ifProp, switchProp } from 'styled-tools';
import { h1, h2, h3, h4, fwSmall, fwMedium, fwLarge } from '../../Style/fonts';
import * as colors from '../../Style/colors';

export default level => {
  const type = `h${level}`;
  return styled[type]`
    ${switchProp('size', {
      h1,
      h2,
      h3,
      h4
    })};

    ${ifProp('center', 'text-align: center;')};

    font-weight: ${switchProp('weight', {
      fwSmall,
      fwMedium,
      fwLarge
    })};

    color: ${props => colors[props.color]};
  `;
};
