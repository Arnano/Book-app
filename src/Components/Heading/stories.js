import React from 'react';
import { storiesOf } from '@storybook/react';
import { wwithInfo, withInfo } from '@storybook/addon-info';
import { withKnobs, text, select, bool } from '@storybook/addon-knobs';

import * as colors from '../../Style/colors';
import { keys } from 'lodash';
import Heading from './';

const selectColors = keys(colors);

const selectSize = {
  h1: 'h1',
  h2: 'h2',
  h3: 'h3',
  h4: 'h4'
};

const selectWeight = {
  fwSmall: 'fwSmall',
  fwMedium: 'fwMedium',
  fwLarge: 'fwLarge'
};

storiesOf('UI Component', module)
  .add('Heading', 
  withInfo({
    text: `
      Heading component accross the application.
      Can be of type h1, h2, h3 or h4
      Color, alignment, font-size and font-weight are customizable.
    `
  }) (() =>
    <Heading
      children={text('Heading: ', 'Heading Sample Text')}
      size={select('Font size: ', selectSize)}
      size={1}
      weight={select('Font weight: ', selectWeight)}
      center={bool('Center: ', false)}
      color={select('Color: ', selectColors)} />
  )
)