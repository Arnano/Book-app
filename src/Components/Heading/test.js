import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Heading from './';

Enzyme.configure({
  adapter: new Adapter()
}); 

describe('Heading', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Heading size='h1' level={1}> Heading </Heading>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Heading size='h1' level={1}> Heading </Heading>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
