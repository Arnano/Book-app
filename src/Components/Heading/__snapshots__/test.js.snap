// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading has a valid snapshot 1`] = `
<h1
  className="sc-bwzfXH hONpvi"
  color="darkGrey"
  id={undefined}
  size="h1"
>
   Heading 
</h1>
`;
